src/hotspot/share/classfile/bytecodeAssembler.cpp

```diff
@@ -33,31 +33,64 @@
 #include "utilities/bytes.hpp"
 #include "utilities/checkedCast.hpp"
 
+void BytecodeConstantPool::init() {
+  for (int i = 1; i < _orig->length(); i++) {
+    bool new_entry = false;
+    BytecodeCPEntry entry;
+    switch(_orig->tag_at(i).value()) {
+    case JVM_CONSTANT_Class:
+    case JVM_CONSTANT_UnresolvedClass:
+      entry = BytecodeCPEntry::klass(_orig->klass_slot_at(i).name_index());
+      break;
+    case JVM_CONSTANT_Utf8:
+      entry = BytecodeCPEntry::utf8(_orig->symbol_at(i));
+      break;
+    case JVM_CONSTANT_NameAndType:
+      entry = BytecodeCPEntry::name_and_type(_orig->name_ref_index_at(i), _orig->signature_ref_index_at(i));
+      break;
+    case JVM_CONSTANT_Methodref:
+      entry = BytecodeCPEntry::methodref(_orig->uncached_klass_ref_index_at(i), _orig->uncached_name_and_type_ref_index_at(i));
+      break;
+    case JVM_CONSTANT_String:
+      entry = BytecodeCPEntry::string(_orig->unresolved_string_at(i));
+      break;
+    }
+    if (entry._tag != BytecodeCPEntry::tag::ERROR_TAG) {
+      bool created = false;
+      u2* probe =_indices.put_if_absent(entry, i, &created);
```

Comment:
Nope. I removed it.  I can't use 'put' instead even though it returns 'created' because 'put' would update the index to this one, not the one it matched, which is sort of a subtle difference.